sim/rogue/combat/combat_potency.go

```diff
@@ -0,0 +1,31 @@
+package combat
+
+import "github.com/wowsims/cata/sim/core"
+
+func (comRogue *CombatRogue) applyCombatPotency() {
+	if comRogue.Talents.CombatPotency == 0 {
+		return
+	}
+
+	const procChance = 0.2
+	energyBonus := 5.0 * float64(comRogue.Talents.CombatPotency)
+	energyMetrics := comRogue.NewEnergyMetrics(core.ActionID{SpellID: 35546})
+
+	comRogue.RegisterAura(core.Aura{
+		Label:    "Combat Potency",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			// from 3.0.3 patch notes: "Combat Potency: Now only works with auto attacks"
+			if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOHAuto) && spell != comRogue.mainGauche {
```

Comment:
OH proc mask but from what I can see mainGauche is a MH spell? Thsi will never trigger

Reply:
This check is inverted from how I'd typically make it (re-used from WotLK implementation). It's saying "If the result did not land, or the spell was a MH hit that wasn't Main Gauche, return".

It's logically correct but a bit of a pain to read. Can invert it if preferred to instead read "If result landed and it's a OH hit or MG hit, continue"

Reply:
Yeah, now that you explain it I see it but it was super confusing on a first read. Up to you honestly how you want it now that I see its correct